src/tiered/tiered_work.c

```diff
@@ -94,6 +94,43 @@ __wt_tiered_pop_work(
     return;
 }
 
+/*
+ * __wt_tiered_flush_work_wait --
+ *     Wait for all flush work units in the work queue to be processed.
+ */
+void
+__wt_tiered_flush_work_wait(WT_SESSION_IMPL *session, uint32_t timeout)
+{
+    struct timespec now, start;
+    WT_CONNECTION_IMPL *conn;
+    WT_TIERED_WORK_UNIT *entry;
+    bool done, found;
+
+    conn = S2C(session);
+    __wt_epoch(session, &start);
+    now = start;
+    done = found = false;
+
+    while (!done) {
+        found = false;
+        __wt_spin_lock(session, &conn->tiered_lock);
+        TAILQ_FOREACH (entry, &conn->tieredqh, q) {
+            if (FLD_ISSET(entry->type, WT_TIERED_WORK_FLUSH))
+                found = true;
+            break;
+        }
+        __wt_spin_unlock(session, &conn->tiered_lock);
+        if (found) {
+            __wt_cond_signal(session, conn->tiered_cond);
+            __wt_sleep(0, 10 * WT_THOUSAND);
+            __wt_epoch(session, &now);
+        }
+        /* We are done if we don't find any work units or exceed the timeout. */
+        done = !found || (WT_TIMEDIFF_SEC(now, start) > timeout);
```

Comment:
It bothers me a bit that this times out silently.  If there's no convenient way to make `WT_CONNECTION->close()` fail outright, we should at least have a message.  It's really likely to happen only when flushing to cloud, but if some current or future tests depend on it, I think we should have some indication.

Reply:
fixed.